inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -616,8 +659,40 @@ std::vector<PointSpeedPair> InLaneCruisingPlugin::maneuvers_to_points(const std:
       points_and_target_speeds.push_back(pair);
     }
   }
+    // Here we are limiting the trajectory length to the given length by maneuver end dist as opposed to the end of lanelets involved.
+    double starting_route_downtrack = wm_->routeTrackPos(points_and_target_speeds.front().point).downtrack; 
+    size_t i = 0;
+    size_t max_i = 0;
+    double dist_accumulator = starting_route_downtrack;
+    lanelet::BasicPoint2d prev_point;
+    for (auto point_speed_pair : points_and_target_speeds) {
+      auto current_point = point_speed_pair.point;
+      if (i == 0) {
+        prev_point = current_point;
+        continue;
+      }
+
+      double delta_d = lanelet::geometry::distance2d(prev_point, current_point);
+      dist_accumulator += delta_d;
+
+      double ending_downtrack = maneuvers.back().lane_following_maneuver.end_dist;
+      if(ending_downtrack + config_.buffer_ending_downtrack < wm_->getRouteEndTrackPos().downtrack){
+        ending_downtrack = ending_downtrack + config_.buffer_ending_downtrack;
+      } 
```

Comment:
This ending downtrack change should be before this for loop as it returns a constant value. No need to have it in the loop.